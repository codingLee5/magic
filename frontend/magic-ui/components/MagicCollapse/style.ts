import { createStyles } from "antd-style"

export const useStyles = createStyles(({ css, prefixCls, token }) => {
	return {
		container: css`
			--${prefixCls}-collapse-content-bg: ${token.colorBgContainer} !important;
			
			.${prefixCls}-collapse-item {
				transition: all 0.35s cubic-bezier(0.215, 0.61, 0.355, 1);
			}
			
			.${prefixCls}-collapse-header {
				transition: all 0.35s cubic-bezier(0.215, 0.61, 0.355, 1) !important;
			}
			
			.${prefixCls}-collapse-content {
				transition: all 0.35s cubic-bezier(0.215, 0.61, 0.355, 1) !important;
			}
		`,
	}
})
