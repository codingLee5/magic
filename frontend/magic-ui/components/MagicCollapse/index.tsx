import { IconChevronRight } from "@tabler/icons-react"
import { Collapse, type CollapseProps } from "antd"
import MagicIcon from "../MagicIcon"
import { useStyles } from "./style"

export type MagicCollapseProps = CollapseProps

function MagicCollapse({ className, ...props }: MagicCollapseProps) {
	const { styles, cx } = useStyles()

	return (
		<Collapse
			ghost
			className={cx(styles.container, className)}
			bordered={false}
			expandIconPosition="end"
			expandIcon={({ isActive }) => (
				<MagicIcon
					component={IconChevronRight}
					size={24}
					style={{
						transform: `rotate(${isActive ? 90 : 0}deg)`,
						transition: "transform 0.35s cubic-bezier(0.215, 0.61, 0.355, 1)",
					}}
				/>
			)}
			{...props}
		/>
	)
}

export default MagicCollapse
