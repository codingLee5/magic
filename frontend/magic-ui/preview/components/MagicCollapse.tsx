import React from "react"
import { Space } from "antd"
import MagicCollapse from "../../components/MagicCollapse"
import ComponentDemo from "./Container"

const CollapseDemo: React.FC = () => {
	const items = [
		{
			key: "1",
			label: "面板1",
			children: "这是面板1的内容，可以包含任何React组件。",
		},
		{
			key: "2",
			label: "面板2",
			children: "这是面板2的内容，支持HTML标签和复杂组件。",
		},
		{
			key: "3",
			label: "面板3",
			children: (
				<div>
					<p>这是面板3的内容，支持复杂的JSX结构。</p>
					<ul>
						<li>列表项1</li>
						<li>列表项2</li>
						<li>列表项3</li>
					</ul>
				</div>
			),
		},
	]

	return (
		<div>
			<ComponentDemo
				title="基础折叠面板"
				description="最基本的折叠面板组件"
				code="<MagicCollapse items={items} />"
			>
				<MagicCollapse items={items} />
			</ComponentDemo>

			<ComponentDemo title="手风琴模式" description="同时只能展开一个面板" code="accordion">
				<MagicCollapse items={items} accordion />
			</ComponentDemo>

			<ComponentDemo
				title="默认展开"
				description="设置默认展开的面板"
				code="defaultActiveKey"
			>
				<MagicCollapse items={items} defaultActiveKey={["1", "2"]} />
			</ComponentDemo>

			<ComponentDemo
				title="受控展开"
				description="通过activeKey控制展开状态"
				code="activeKey | onChange"
			>
				<MagicCollapse
					items={items}
					activeKey={["1"]}
					onChange={(keys) => console.log("展开的面板:", keys)}
				/>
			</ComponentDemo>

			<ComponentDemo
				title="不同尺寸"
				description="支持不同尺寸的折叠面板"
				code="size: 'large' | 'default' | 'small'"
			>
				<Space direction="vertical" style={{ width: "100%" }}>
					<MagicCollapse items={items} size="large" />
					<MagicCollapse items={items} />
					<MagicCollapse items={items} size="small" />
				</Space>
			</ComponentDemo>
		</div>
	)
}

export default CollapseDemo
