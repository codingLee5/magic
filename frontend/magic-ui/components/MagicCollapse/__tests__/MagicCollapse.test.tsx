import { render, screen } from "@testing-library/react"
import MagicThemeProvider from "../../ThemeProvider"
import { Collapse } from "antd"
import { describe, it, expect } from "vitest"
import MagicCollapse from "../index"

const renderWithTheme = (component: React.ReactElement) =>
	render(<MagicThemeProvider theme="light">{component}</MagicThemeProvider>)

describe("MagicCollapse", () => {
	it("应该正常渲染", () => {
		renderWithTheme(
			<MagicCollapse>
				<Collapse.Panel key="1" header="标题1">
					内容1
				</Collapse.Panel>
			</MagicCollapse>,
		)
		expect(screen.getByText("标题1")).toBeInTheDocument()
	})

	// 快照测试
	describe("快照测试", () => {
		it("基础折叠面板快照", () => {
			const { asFragment } = renderWithTheme(
				<MagicCollapse>
					<Collapse.Panel key="1" header="标题1">
						内容1
					</Collapse.Panel>
				</MagicCollapse>,
			)
			expect(asFragment()).toMatchSnapshot()
		})

		it("多个面板折叠快照", () => {
			const { asFragment } = renderWithTheme(
				<MagicCollapse>
					<Collapse.Panel key="1" header="标题1">
						内容1
					</Collapse.Panel>
					<Collapse.Panel key="2" header="标题2">
						内容2
					</Collapse.Panel>
					<Collapse.Panel key="3" header="标题3">
						内容3
					</Collapse.Panel>
				</MagicCollapse>,
			)
			expect(asFragment()).toMatchSnapshot()
		})

		it("默认展开面板快照", () => {
			const { asFragment } = renderWithTheme(
				<MagicCollapse defaultActiveKey={["1"]}>
					<Collapse.Panel key="1" header="标题1">
						内容1
					</Collapse.Panel>
					<Collapse.Panel key="2" header="标题2">
						内容2
					</Collapse.Panel>
				</MagicCollapse>,
			)
			expect(asFragment()).toMatchSnapshot()
		})

		it("手风琴模式快照", () => {
			const { asFragment } = renderWithTheme(
				<MagicCollapse accordion>
					<Collapse.Panel key="1" header="标题1">
						内容1
					</Collapse.Panel>
					<Collapse.Panel key="2" header="标题2">
						内容2
					</Collapse.Panel>
				</MagicCollapse>,
			)
			expect(asFragment()).toMatchSnapshot()
		})

		it("带图标面板快照", () => {
			const { asFragment } = renderWithTheme(
				<MagicCollapse>
					<Collapse.Panel key="1" header="标题1" extra={<span>📝</span>}>
						内容1
					</Collapse.Panel>
				</MagicCollapse>,
			)
			expect(asFragment()).toMatchSnapshot()
		})

		it("禁用面板快照", () => {
			const { asFragment } = renderWithTheme(
				<MagicCollapse>
					<Collapse.Panel key="1" header="标题1" disabled>
						内容1
					</Collapse.Panel>
				</MagicCollapse>,
			)
			expect(asFragment()).toMatchSnapshot()
		})

		it("小尺寸折叠面板快照", () => {
			const { asFragment } = renderWithTheme(
				<MagicCollapse size="small">
					<Collapse.Panel key="1" header="标题1">
						内容1
					</Collapse.Panel>
				</MagicCollapse>,
			)
			expect(asFragment()).toMatchSnapshot()
		})
	})
})
