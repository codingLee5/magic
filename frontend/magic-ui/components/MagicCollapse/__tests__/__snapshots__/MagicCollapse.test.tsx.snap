// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MagicCollapse > 快照测试 > 基础折叠面板快照 1`] = `
<DocumentFragment>
  <div
    class="magic-collapse magic-collapse-icon-position-end magic-collapse-borderless magic-collapse-ghost acss-1up74ee css-dev-only-do-not-override-yotw4n css-var-r1"
  >
    <div
      class="magic-collapse-item"
    >
      <div
        aria-disabled="false"
        aria-expanded="false"
        class="magic-collapse-header"
        role="button"
        tabindex="0"
      >
        <div
          class="magic-collapse-expand-icon"
        >
          <svg
            class="tabler-icon tabler-icon-chevron-right magic-collapse-arrow"
            fill="none"
            height="24"
            stroke="rgba(28,29,35, 0.8)"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
            style="transform: rotate(0deg); transition: transform 0.35s cubic-bezier(0.215, 0.61, 0.355, 1);"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 6l6 6l-6 6"
            />
          </svg>
        </div>
        <span
          class="magic-collapse-header-text"
        >
          标题1
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MagicCollapse > 快照测试 > 多个面板折叠快照 1`] = `
<DocumentFragment>
  <div
    class="magic-collapse magic-collapse-icon-position-end magic-collapse-borderless magic-collapse-ghost acss-1up74ee css-dev-only-do-not-override-yotw4n css-var-r2"
  >
    <div
      class="magic-collapse-item"
    >
      <div
        aria-disabled="false"
        aria-expanded="false"
        class="magic-collapse-header"
        role="button"
        tabindex="0"
      >
        <div
          class="magic-collapse-expand-icon"
        >
          <svg
            class="tabler-icon tabler-icon-chevron-right magic-collapse-arrow"
            fill="none"
            height="24"
            stroke="rgba(28,29,35, 0.8)"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
            style="transform: rotate(0deg); transition: transform 0.35s cubic-bezier(0.215, 0.61, 0.355, 1);"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 6l6 6l-6 6"
            />
          </svg>
        </div>
        <span
          class="magic-collapse-header-text"
        >
          标题1
        </span>
      </div>
    </div>
    <div
      class="magic-collapse-item"
    >
      <div
        aria-disabled="false"
        aria-expanded="false"
        class="magic-collapse-header"
        role="button"
        tabindex="0"
      >
        <div
          class="magic-collapse-expand-icon"
        >
          <svg
            class="tabler-icon tabler-icon-chevron-right magic-collapse-arrow"
            fill="none"
            height="24"
            stroke="rgba(28,29,35, 0.8)"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
            style="transform: rotate(0deg); transition: transform 0.35s cubic-bezier(0.215, 0.61, 0.355, 1);"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 6l6 6l-6 6"
            />
          </svg>
        </div>
        <span
          class="magic-collapse-header-text"
        >
          标题2
        </span>
      </div>
    </div>
    <div
      class="magic-collapse-item"
    >
      <div
        aria-disabled="false"
        aria-expanded="false"
        class="magic-collapse-header"
        role="button"
        tabindex="0"
      >
        <div
          class="magic-collapse-expand-icon"
        >
          <svg
            class="tabler-icon tabler-icon-chevron-right magic-collapse-arrow"
            fill="none"
            height="24"
            stroke="rgba(28,29,35, 0.8)"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
            style="transform: rotate(0deg); transition: transform 0.35s cubic-bezier(0.215, 0.61, 0.355, 1);"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 6l6 6l-6 6"
            />
          </svg>
        </div>
        <span
          class="magic-collapse-header-text"
        >
          标题3
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MagicCollapse > 快照测试 > 小尺寸折叠面板快照 1`] = `
<DocumentFragment>
  <div
    class="magic-collapse magic-collapse-icon-position-end magic-collapse-borderless magic-collapse-ghost magic-collapse-small acss-1up74ee css-dev-only-do-not-override-yotw4n css-var-r7"
  >
    <div
      class="magic-collapse-item"
    >
      <div
        aria-disabled="false"
        aria-expanded="false"
        class="magic-collapse-header"
        role="button"
        tabindex="0"
      >
        <div
          class="magic-collapse-expand-icon"
        >
          <svg
            class="tabler-icon tabler-icon-chevron-right magic-collapse-arrow"
            fill="none"
            height="24"
            stroke="rgba(28,29,35, 0.8)"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
            style="transform: rotate(0deg); transition: transform 0.35s cubic-bezier(0.215, 0.61, 0.355, 1);"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 6l6 6l-6 6"
            />
          </svg>
        </div>
        <span
          class="magic-collapse-header-text"
        >
          标题1
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MagicCollapse > 快照测试 > 带图标面板快照 1`] = `
<DocumentFragment>
  <div
    class="magic-collapse magic-collapse-icon-position-end magic-collapse-borderless magic-collapse-ghost acss-1up74ee css-dev-only-do-not-override-yotw4n css-var-r5"
  >
    <div
      class="magic-collapse-item"
    >
      <div
        aria-disabled="false"
        aria-expanded="false"
        class="magic-collapse-header"
        role="button"
        tabindex="0"
      >
        <div
          class="magic-collapse-expand-icon"
        >
          <svg
            class="tabler-icon tabler-icon-chevron-right magic-collapse-arrow"
            fill="none"
            height="24"
            stroke="rgba(28,29,35, 0.8)"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
            style="transform: rotate(0deg); transition: transform 0.35s cubic-bezier(0.215, 0.61, 0.355, 1);"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 6l6 6l-6 6"
            />
          </svg>
        </div>
        <span
          class="magic-collapse-header-text"
        >
          标题1
        </span>
        <div
          class="magic-collapse-extra"
        >
          <span>
            📝
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MagicCollapse > 快照测试 > 手风琴模式快照 1`] = `
<DocumentFragment>
  <div
    class="magic-collapse magic-collapse-icon-position-end magic-collapse-borderless magic-collapse-ghost acss-1up74ee css-dev-only-do-not-override-yotw4n css-var-r4"
    role="tablist"
  >
    <div
      class="magic-collapse-item"
    >
      <div
        aria-disabled="false"
        aria-expanded="false"
        class="magic-collapse-header"
        role="tab"
        tabindex="0"
      >
        <div
          class="magic-collapse-expand-icon"
        >
          <svg
            class="tabler-icon tabler-icon-chevron-right magic-collapse-arrow"
            fill="none"
            height="24"
            stroke="rgba(28,29,35, 0.8)"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
            style="transform: rotate(0deg); transition: transform 0.35s cubic-bezier(0.215, 0.61, 0.355, 1);"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 6l6 6l-6 6"
            />
          </svg>
        </div>
        <span
          class="magic-collapse-header-text"
        >
          标题1
        </span>
      </div>
    </div>
    <div
      class="magic-collapse-item"
    >
      <div
        aria-disabled="false"
        aria-expanded="false"
        class="magic-collapse-header"
        role="tab"
        tabindex="0"
      >
        <div
          class="magic-collapse-expand-icon"
        >
          <svg
            class="tabler-icon tabler-icon-chevron-right magic-collapse-arrow"
            fill="none"
            height="24"
            stroke="rgba(28,29,35, 0.8)"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
            style="transform: rotate(0deg); transition: transform 0.35s cubic-bezier(0.215, 0.61, 0.355, 1);"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 6l6 6l-6 6"
            />
          </svg>
        </div>
        <span
          class="magic-collapse-header-text"
        >
          标题2
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MagicCollapse > 快照测试 > 禁用面板快照 1`] = `
<DocumentFragment>
  <div
    class="magic-collapse magic-collapse-icon-position-end magic-collapse-borderless magic-collapse-ghost acss-1up74ee css-dev-only-do-not-override-yotw4n css-var-r6"
  >
    <div
      class="magic-collapse-item magic-collapse-item-disabled"
      disabled=""
    >
      <div
        aria-disabled="true"
        aria-expanded="false"
        class="magic-collapse-header magic-collapse-collapsible-disabled"
        role="button"
        tabindex="-1"
      >
        <div
          class="magic-collapse-expand-icon"
        >
          <svg
            class="tabler-icon tabler-icon-chevron-right magic-collapse-arrow"
            fill="none"
            height="24"
            stroke="rgba(28,29,35, 0.8)"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
            style="transform: rotate(0deg); transition: transform 0.35s cubic-bezier(0.215, 0.61, 0.355, 1);"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 6l6 6l-6 6"
            />
          </svg>
        </div>
        <span
          class="magic-collapse-header-text"
        >
          标题1
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MagicCollapse > 快照测试 > 默认展开面板快照 1`] = `
<DocumentFragment>
  <div
    class="magic-collapse magic-collapse-icon-position-end magic-collapse-borderless magic-collapse-ghost acss-1up74ee css-dev-only-do-not-override-yotw4n css-var-r3"
  >
    <div
      class="magic-collapse-item magic-collapse-item-active"
    >
      <div
        aria-disabled="false"
        aria-expanded="true"
        class="magic-collapse-header"
        role="button"
        tabindex="0"
      >
        <div
          class="magic-collapse-expand-icon"
        >
          <svg
            class="tabler-icon tabler-icon-chevron-right magic-collapse-arrow"
            fill="none"
            height="24"
            stroke="rgba(28,29,35, 0.8)"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
            style="transform: rotate(90deg); transition: transform 0.35s cubic-bezier(0.215, 0.61, 0.355, 1);"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 6l6 6l-6 6"
            />
          </svg>
        </div>
        <span
          class="magic-collapse-header-text"
        >
          标题1
        </span>
      </div>
      <div
        class="magic-collapse-content magic-collapse-content-active"
      >
        <div
          class="magic-collapse-content-box"
        >
          内容1
        </div>
      </div>
    </div>
    <div
      class="magic-collapse-item"
    >
      <div
        aria-disabled="false"
        aria-expanded="false"
        class="magic-collapse-header"
        role="button"
        tabindex="0"
      >
        <div
          class="magic-collapse-expand-icon"
        >
          <svg
            class="tabler-icon tabler-icon-chevron-right magic-collapse-arrow"
            fill="none"
            height="24"
            stroke="rgba(28,29,35, 0.8)"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
            style="transform: rotate(0deg); transition: transform 0.35s cubic-bezier(0.215, 0.61, 0.355, 1);"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 6l6 6l-6 6"
            />
          </svg>
        </div>
        <span
          class="magic-collapse-header-text"
        >
          标题2
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
